example.go

```diff
@@ -26,29 +33,21 @@ func main() {
 	verboseFlag := flag.Bool("v", false, "Verbose mode, which prints raw server data")
 	flag.Parse()
 
-	clientID := *clientIDFlag
-	clientKey := *clientKeyFlag
+	// Make log not print out time info
+	log.SetFlags(0)
 
-	if clientID == "" {
-		// check environment variable
-		if os.Getenv("HOUNDIFY_CLIENT_ID") != "" {
-			clientID = os.Getenv("HOUNDIFY_CLIENT_ID")
-		}
-	}
-	if clientKey == "" {
-		// check environment variable
-		if os.Getenv("HOUNDIFY_CLIENT_KEY") != "" {
-			clientKey = os.Getenv("HOUNDIFY_CLIENT_KEY")
-		}
-	}
+	clientID := derefOrFetchFromEnv(clientIDFlag, envClientIDKey)
+	clientKey := derefOrFetchFromEnv(clientKeyFlag, envClientKeyKey)
 
+	var errsList []string
 	if clientID == "" {
-		fmt.Println("Need to set the client ID")
-		os.Exit(1)
+		errsList = append(errsList, fmt.Sprintf("must set the client ID in environment variable: %q", envClientIDKey))
```

Comment:
Could this error message be modified to also mention command line flag option too, something like "must set the client ID in the environment variable %q or set via the command line flag -%s"